test/libsolidity/Builtin.h

```diff
@@ -0,0 +1,55 @@
+/*
+	This file is part of solidity.
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include <functional>
+#include <map>
+#include <optional>
+#include <test/libsolidity/util/SoltestErrors.h>
+#include <test/libsolidity/util/SoltestTypes.h>
+#include <utility>
+#include <vector>
+#include <memory>
+
+namespace solidity::frontend::test
+{
+
+class TestFunctionCall;
+
+class Builtin
+{
+	/// This is the actual implementation of the builtin function.
+	std::function<std::optional<bytes>(FunctionCall const&)> m_builtin = nullptr;
```

Comment:
Does this need to be initialized? Can it be initialized with nullptr at all?

Reply:
Hmm.. basically I see this just like a function pointer. I just initialise it to nullptr to know whether it got set to something or not. I guess it is not really needed to get initialised here, but I think it's better to be explicit here.

Reply:
I'm not sure if you need to initialize it, but it would probably be better to use `= {}` or something. It might actually be that `nullptr` is different from the default value.

Reply:
Ok, I removed the implicit initialialization.